system/Cache/CacheInterface.php

```diff
@@ -11,6 +11,8 @@
 
 namespace CodeIgniter\Cache;
 
+use Closure;
+
```

Comment:
I think this is unused.

Reply:
It is not used, but I'm torn on whether it should be. Adding a method to the handlers without indicating it in the interface makes it unusable by anything type-hinted `CacheInterface`. However, adding it to the interface is a breaking change. This is why I tend to favor abstract base classes over interfaces.

Maybe since we are adding `BaseHandler` to consolidate `remember()` we should move these definitions there as abstract methods and deprecate the whole interface?

Reply:
I am against deprecating the interface. An abstract class, as an incomplete implementation of a contract, can provide the default implementations of the contract or add new features on top of it. So an abstract class is a supplement to an interface, not its replacement.

Another way I'm thinking is extending the CacheInterface by a new interface with this method `remember`. Then we can either change the phpdocs to use this new interface or use `instanceof` checks.
`($cache instanceof NewInterface) && $cache->remember();`